blockchain/tests/inherents.rs

```diff
@@ -699,3 +699,119 @@ async fn create_fork_proof() {
     // Verify that the fork proof was generated
     assert!(fork_rx.next().await.is_some());
 }
+
+#[test]
+fn it_can_create_version_upgrade_inherents() {
+    let time = Arc::new(OffsetTime::new());
+    let env = MdbxDatabase::new_volatile(Default::default()).unwrap();
+    let blockchain = Arc::new(
+        Blockchain::new(
+            env,
+            BlockchainConfig::default(),
+            NetworkId::UnitAlbatross,
+            time,
+        )
+        .unwrap(),
+    );
+
+    let block_number = Policy::election_block_after(Policy::genesis_block_number());
+
+    let staking_contract = blockchain.get_staking_contract();
+    let active_validators = staking_contract.active_validators.clone();
+    let next_batch_initial_punished_set = staking_contract
+        .punished_slots
+        .next_batch_initial_punished_set(block_number, &active_validators);
+
+    let mut macro_header = MacroHeader {
+        network: NetworkId::UnitAlbatross,
+        version: 2,
+        block_number,
+        round: 0,
+        timestamp: blockchain.state.election_head.header.timestamp + 20000,
+        parent_hash: Blake2bHash::default(),
+        parent_election_hash: Blake2bHash::default(),
+        interlink: None,
+        seed: VrfSeed::default(),
+        extra_data: vec![],
+        state_root: Blake2bHash::default(),
+        body_root: Blake2sHash::default(),
+        diff_root: Blake2bHash::default(),
+        history_root: Blake2bHash::default(),
+        validators: None,
+        next_batch_initial_punished_set,
+        ..Default::default()
+    };
```

Comment:
Most of this struct can probably be omitted since you specify `..Default::default()` anyway.